tchannel/tornado/dispatch.py

```diff
@@ -220,20 +222,22 @@ def handle_call(self, request, connection):
 
             response.flush()
         except TChannelError as e:
-            connection.send_error(
-                e.code,
-                e.message,
-                request.id,
-            )
+            e.tracing = request.tracing
+            e.id = request.id
+            connection.send_error(e)
         except Exception as e:
-            msg = "An unexpected error has occurred from the handler"
-            log.exception(msg)
+            log.exception("Unexpected Error: {0}".format(e.message))
 
-            response.set_exception(TChannelError(e.message))
+            error = UnexpectedError(
+                description='An unexpected error occurred from the handler',
+                id=request.id,
+                tracing=request.tracing,
```

Comment:
What's going on with all the tracing stuff? Seems like that's a real bugfix that belongs in its own patch with its own tests.